spec/datadog/core/telemetry/request_spec.rb

```diff
@@ -3,6 +3,14 @@
 require 'datadog/core/telemetry/request'
 
 RSpec.describe Datadog::Core::Telemetry::Request do
+  before do
+    # The tests here assert on time equality after rounding to a second;
+    # these assertions will fail if the code executes just before
+    # a second boundary and the assertions run just after the second boundary.
+    # Align the runs closer to the beginning of a second to reduce flakiness.
+    sleep 0 until Time.now.usec < 750000
+  end
```

Comment:
In the profiler tests I have a few similar situations where I don't know exactly the expected timestamp.

My suggestion here, rather than the sleep/loop (which doesn't exactly guarantee the timings we need -- just makes it more likely to happen) would be to do something like:

```ruby
let!(:before_time) { Time.now.to_i }

# ...

    it do
      is_expected.to eq(
        api_version: api_version,
        application: application,
        debug: debug,
        host: host,
        payload: payload,
        request_type: request_type,
        runtime_id: runtime_id,
        seq_id: seq_id,
        tracer_time: be_between(before_time, Time.now.to_i),
      )
    end
```

This way we validate the timestamp is correct without the loop or any flakiness.

Reply:
I implemented this in https://github.com/DataDog/dd-trace-rb/pull/3966.